File: legacy/pasarguardbot/app/models/panel/panel_options.py
from pydantic import BaseModel, Field

from app.models.panel.common import PagedRequest, PageMeta, PanelResponse

PANEL_OPTION_FIELDS = ("code", "name", "enable", "base_url")
DEFAULT_PANEL_OPTION_FIELDS = ("code", "name")


class PanelOptionsRequest(PagedRequest):
    q: str = Field("", max_length=64)
    fields: list[str] = Field(default_factory=lambda: list(DEFAULT_PANEL_OPTION_FIELDS))


class PanelOptionRow(BaseModel):
    code: int
    name: str | None = None
    enable: bool | None = None
    base_url: str | None = None


class PanelOptionsResponse(PanelResponse):
    panels: list[PanelOptionRow] = Field(default_factory=list)
    meta: PageMeta = Field(default_factory=PageMeta)

File: legacy/pasarguardbot/frontend/src/types/panel/panel_options.ts
/** Mirrors app/models/panel/panel_options.py */
import type { PagedRequest, PanelEnvelope, PageMeta } from "./common";

/** Never add secret-bearing columns here (password, cookie) — the backend
 * whitelists this exact same set and ignores anything else. */
export const PANEL_OPTION_FIELDS = ["code", "name", "enable", "base_url"] as const;

export interface PanelOptionsRequest extends PagedRequest {
  q?: string;
  /** Subset of PANEL_OPTION_FIELDS; defaults to ["code", "name"]. */
  fields?: string[];
}

export interface PanelOptionRow {
  code: number;
  name?: string | null;
  enable?: boolean | null;
  base_url?: string | null;
}

export interface PanelOptionsResponse extends PanelEnvelope {
  panels: PanelOptionRow[];
  meta: PageMeta;
}
